src/de/uni_passau/fim/gitwrapper/EventDataProcessor.java

```diff
@@ -43,7 +45,18 @@ class EventDataProcessor implements JsonDeserializer<EventData>, JsonSerializer<
         map.put("unlabeled", EventData.LabeledEventData.class);
         map.put("referenced", EventData.ReferencedEventData.class);
         map.put("merged", EventData.ReferencedEventData.class);
-        map.put("closed", EventData.ReferencedEventData.class);
+        map.put("closed", EventData.StateChangedEventData.class);
+        map.put("connected", EventData.ConnectedEventData.class);
+        map.put("reopened", EventData.StateChangedEventData.class);
+        map.put("issue_type_added", EventData.IssueTypeChangedEventData.class);
+        map.put("issue_type_changed", EventData.IssueTypeChangedEventData.class);
+        map.put("issue_type_removed", EventData.IssueTypeChangedEventData.class);
+        map.put("parent_issue_added", EventData.ParentIssueChangedEventData.class);
+        map.put("parent_issue_removed", EventData.ParentIssueChangedEventData.class);
+        map.put("parent_issue_changed", EventData.ParentIssueChangedEventData.class);
+        map.put("sub_issue_added", EventData.SubIssueChangedEventData.class);
+        map.put("sub_issue_removed", EventData.SubIssueChangedEventData.class);
+        map.put("sub_issue_changed", EventData.SubIssueChangedEventData.class);
```

Comment:
If I remember correctly, a couple of weeks ago, we had a discussion in which you @Leo-Send mentioned that some/one of these "changed" events does not exist (not sure whether it was issue type, parent issue, or sub issue). Could you please check that again and remove the non-existing events here and everywhere else if applicable?